utils/src/types/unsigned_tx.rs

```diff
@@ -0,0 +1,521 @@
+use super::*;
+use crate::buffer::{Buffer, Writable};
+use crate::decode::*;
+
+#[cfg_attr(test, derive(Debug))]
+pub enum UnsignedTx {
```

Comment:
Q: Should `UnsignedTx` be a struct?

Reply:
Same as above, the purpose is to reduce the stack memory size. In rust, if we don't consider the memory alignment, the size of a struct is the sum of all struct fields, but the size of an enum is the largest enum fields.